pkg/statistics/histogram.go

```diff
@@ -1049,22 +1073,26 @@ func (hg *Histogram) OutOfRangeRowCount(
 		upperBound = hg.NotNullCount() / float64(histNDV)
 	}
 
-	allowUseModifyCount := sctx.GetSessionVars().GetOptObjective() != variable.OptObjectiveDeterminate
-
 	if !allowUseModifyCount {
 		// In OptObjectiveDeterminate mode, we can't rely on the modify count anymore.
 		// An upper bound is necessary to make the estimation make sense for predicates with bound on only one end, like a > 1.
 		// We use 1/NDV here (only the Histogram part is considered) and it seems reasonable and good enough for now.
 		return min(rowCount, upperBound)
 	}
 
-	// If the modifyCount is large (compared to original table rows), then any out of range estimate is unreliable.
+	// If the realtimeRowCount is larger than the original table rows, then any out of range estimate is unreliable.
```

Comment:
Can you give an example of where this change would help? I can understand the changes, but I don't understand the impact of this change. Forgive me for asking too many questions. I just want to make sure I understand the motivation behind this change.

Reply:
Simple example - if I have a table with 100 rows, and values 1 to 100.
Select count(*) from t where a > 100; <-- will return count == 0.

If I insert 10 rows, values 101 to 110 - and do NOT run ANALYZE - realtimeRowCount == 110.

Select count(*) from t where a > 100; <-- now the result is count == 10 (because I added 10 values after 100.

Now I run ANALYZE (so rowCount == 110), then insert another 10 rows - this time with values 1-10 - realtimeRowCount == 120.
select count(*) from t where a > 110; <-- realtimeRowCount == 120 - and I don't know that the 10 new rows aren't greater than 110.